dozer-cli/src/simple/executor.rs

```diff
@@ -116,11 +116,12 @@ impl<'a> Executor<'a> {
 }
 
 pub fn run_dag_executor(
+    runtime: &Runtime,
     dag_executor: DagExecutor,
     running: Arc<AtomicBool>,
     labels: LabelsAndProgress,
 ) -> Result<(), OrchestrationError> {
-    let join_handle = dag_executor.start(running, labels)?;
+    let join_handle = runtime.block_on(dag_executor.start(running, labels))?;
```

Comment:
This function runs in `spawn_blocking` because `join_handle.join` is blocking (it's joining threads not futures). If we `await` or `block_on` at the caller, async runtime is blocked.

Reply:
Ah, `dag_executor.start()` returns a future that resolves to a thread join handle.